src/commands/constelBundlesFiltering/bundlesFiltering.cc

```diff
@@ -141,83 +147,186 @@ int main(int argc, const char* argv[]) {
 
     unsigned i, n = fileNameIn.size();
 
-    // create common elements
-
-    vector<rc_ptr<BundleReader> > bundle(n);
-    vector<rc_ptr<SelectFiberListenerFromMesh> > gyriFilter(n);
-
-    vector<rc_ptr<SelectBundlesFromNames> > selectCortexBundles(n);
-    vector<rc_ptr<SelectBundlesFromLength> > selectCBundlesFromLength(n);
-    // regroup cortex bundles by gyrus name
-    rc_ptr<BundlesFusion> cortexRegroup(new BundlesFusion((int) n));
-
-    vector<rc_ptr<SelectBundlesFromNames> > selectNimBundles(n);
-    vector<rc_ptr<SelectBundlesFromLength> > selectNBundlesFromLength(n);
-    // regroup "not in mesh" bundles by gyrus name
-    rc_ptr<BundlesFusion> nimRegroup( new BundlesFusion((int) n));
-
-    // duplicate branches for all input files
-    for (i=0; i!=n; ++i) {
-      // Bundles reader creation
-      string fileName = fileNameIn[i];
-      bundle[i].reset( new BundleReader( fileName ) );
-
-      // Set names from mesh/label texture
-      gyriFilter[i].reset(
-        new SelectFiberListenerFromMesh(mesh, tex, mode, addInt, motion,""));
-      bundle[i]->addBundleListener(*gyriFilter[i]);
-
-      // -- 1st branch: near cortex
-      // filter labels
-      selectCortexBundles[i].reset(
-        new SelectBundlesFromNames(namesList, verbose, as_regex, true));
-      gyriFilter[i]->addBundleListener(*selectCortexBundles[i]);
-
-      // filter from length
-      selectCBundlesFromLength[i].reset(
-        new SelectBundlesFromLength(cortMinlength, cortMaxlength, verbose));
-      selectCortexBundles[i]->addBundleListener(*selectCBundlesFromLength[i]);
-
-      // connect to common cortex fusion element
-      selectCBundlesFromLength[i]->addBundleListener(*cortexRegroup);
-
-      // -- 2nd branch: not in mesh
-      // filter labels
-      vector<string> notinmesh_names;
-      notinmesh_names.push_back(gyrus + "_notInMesh");
-      selectNimBundles[i].reset( 
-        new SelectBundlesFromNames(notinmesh_names, verbose, false, true));
-      gyriFilter[i]->addBundleListener(*selectNimBundles[i]);
-
-      // filter from length
-      selectNBundlesFromLength[i].reset(
-        new SelectBundlesFromLength(nimMinlength, nimMaxlength, verbose));
-      selectNimBundles[i]->addBundleListener(*selectNBundlesFromLength[i]);
-
-      // regroup bundles by gyrus name
-      selectNBundlesFromLength[i]->addBundleListener(*nimRegroup);
+    // MRtrix pipeline
```

Comment:
Why is the whole pipeline duplicated ? Wasn't it just a matter of adding a BundleWeightReader in the existing chains ?
Because the case of multiple bundles files is not handled, is it ?